pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/LogIndexBackoff.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.transaction;
+
+import org.apache.pulsar.client.api.RedeliveryBackoff;
+
+public class LogIndexBackoff implements RedeliveryBackoff {
+
+    private final long minLag;
+    private final long maxLag;
+
+    public LogIndexBackoff(long minLag, long maxLag) {
+        this.minLag = minLag;
+        this.maxLag = maxLag;
+    }
+
+    public long getMinLag() {
+        return this.minLag;
+    }
+
+    public long getMaxLag() {
+        return this.maxLag;
+    }
+
+    @Override
+    public long next(int indexCount) {
+        if (indexCount <= 0 || minLag <= 0) {
+            return this.minLag;
```

Comment:
may this should return 0

Reply:
I have also thought about this. I don't think there should be two indexes of contiguous positions stored in pendingAck.
In addition to the initial value minLag should be 0, minlag should not be 0 after that.